containerd-shim-v2/service.go

```diff
@@ -337,34 +337,46 @@ func (s *service) Create(ctx context.Context, r *taskAPI.CreateTaskRequest) (_ *
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	var c *container
-
-	c, err = create(ctx, s, r)
-	if err != nil {
-		return nil, err
-	}
-
-	c.status = task.StatusCreated
-
-	s.containers[r.ID] = c
+	type Result struct {
+		container *container
+		err       error
+	}
+	ch := make(chan Result, 1)
+	go func() {
+		container, err := create(ctx, s, r)
```

Comment:
I think one thing that you need to be careful of is to make sure that calls within this `create` func handles context.Cancel() correctly.  Running this in a goroutine is fine but you run the risks of this hanging forever on a cancel if the calls within this func do not also handle a canceled context.

Reply:
I suppose the original code doesn't handle cancel ctx as well? The fact is this `Create` service does NOT handle any ctx at all... :(
So do you think this goroutinue will make it more risky?

Reply:
I guess since this is a create, a hanging go routine is fine as the shim will be shutdown following the error.

I think working on the create to fully handle the context can come in a separate PR to make this even more robust :)

Reply:
ping @crosbymichael
Anything left for me to do? :)

Reply:
nope, all good